jsonschema2md/__init__.py

```diff
@@ -590,11 +632,76 @@ def _parse_object(
 
         return output_lines
 
+    def parse_file(
+        self,
+        file: Path,
+        fail_on_error_in_defs: bool = True,
+        locale: Optional[str] = None,
+    ) -> dict[str, Sequence[str]]:
+        """
+        Parse JSON Schema file and its references to Markdown text.
+
+        Parameters
+        ----------
+        file: Path
+            The Path to the JSON Schema file to parse.
+        fail_on_error_in_defs: bool
+            If True, the method will raise an error when encountering issues in the
+            "definitions" section of the schemas. If False, the method will attempt to continue parsing
+            despite such errors.
+        locale: Optional[str]
+            The locale to use for translations. If None, the default locale will be used.
+
+        Returns
+        -------
+        dict[str, Sequence[str]]
+            A dictionary where keys are file names (without `.json` extension) and values are lists of strings
+            representing the parsed Markdown documentation for each file.
+        """
+        if locale is not None:
+            Parser.current_locale = negotiate_locale((locale,), get_locales())
+
+        with file.open(encoding="utf-8") as input_file:
+            schema_obj = json.load(input_file)
+
+        root = self.parse_schema(schema_obj, fail_on_error_in_defs)
+        parsed_files = {"_js2md_root_": root}
+
+        if self.domain:
+            for _ in range(3):
+                to_parse = self.seen_refs - self.parsed_refs
+                if not to_parse:
+                    break
+
+                for ref in to_parse:
+                    ref_file = Path(f"{file}/../{ref}").resolve()
```

Comment:
Better :-)
```suggestion
                    ref_file = (file.parent / ref).resolve()
```
And I think that the resolve is not needed :-)